src/Mod/Material/App/Materials.cpp

```diff
@@ -448,6 +448,17 @@ bool MaterialProperty::operator==(const MaterialProperty& other) const
     return false;
 }
 
+void MaterialProperty::validate(const MaterialProperty& other) const {
+    _valuePtr->validate(*other._valuePtr);
+
+    if (_columns.size() != other._columns.size()) {
+        throw InvalidProperty("Model property column counts don't match");
+    }
+    for (int i = 0; i < _columns.size(); i++) {
```

Comment:
Compiler warning: signed to unsigned comparison
```suggestion
    for (size_t i = 0; i < _columns.size(); i++) {
```